homeassistant/components/keymitt_ble/config_flow.py

```diff
@@ -0,0 +1,156 @@
+
+import logging
+from typing import Any
+
+from bleak.backends.device import BLEDevice
+from microbot import (
+    MicroBotAdvertisement,
+    MicroBotApiClient,
+    parse_advertisement_data,
+    randomid,
+)
+import voluptuous as vol
+
+from homeassistant.components import bluetooth
+from homeassistant.components.bluetooth import (
+    BluetoothServiceInfoBleak,
+    async_discovered_service_info,
+)
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.const import CONF_ACCESS_TOKEN
+
+from .const import CONF_BDADDR, DOMAIN, DEFAULT_NAME
+
+_LOGGER: logging.Logger = logging.getLogger(__package__)
+
+
+def short_address(address: str) -> str:
+    """Convert a Bluetooth address to a short address."""
+    results = address.replace("-", ":").split(":")
+    return f"{results[0].upper()}{results[1].upper()}"[0:4]
+
+
+def name_from_discovery(discovery: MicroBotAdvertisement) -> str:
+    """Get the name from a discovery."""
+    return f'{discovery.data["local_name"]} {short_address(discovery.address)}'
+
+
+class MicroBotConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Config flow for MicroBot."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize."""
+        self._errors = {}
+        self._discovered_adv: MicroBotAdvertisement | None = None
+        self._discovered_advs: dict[str, MicroBotAdvertisement] = {}
+        self._client: MicroBotApiClient | None = None
+        self._ble_device: BLEDevice | None
+        self._name = None
+        self._bdaddr: str
+
+    async def async_step_bluetooth(
+        self, discovery_info: BluetoothServiceInfoBleak
+    ) -> FlowResult:
+        """Handle the bluetooth discovery step."""
+        _LOGGER.debug("Discovered bluetooth device: %s", discovery_info)
+        await self.async_set_unique_id(discovery_info.address)
+        self._abort_if_unique_id_configured()
+        self._ble_device = discovery_info.device
+        parsed = parse_advertisement_data(
+            discovery_info.device, discovery_info.advertisement
+        )
+        self._discovered_adv = parsed
+        self.context["title_placeholders"] = {
+            "name": name_from_discovery(self._discovered_adv),
+        }
+        return await self.async_step_init()
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        # This is for backwards compatibility.
+        return await self.async_step_init(user_input)
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Check if paired."""
+        errors: dict[str, str] = {}
+
+        if discovery := self._discovered_adv:
+            self._discovered_advs[discovery.address] = discovery
+            self._name = name_from_discovery(discovery)
+        else:
+            self._name = DEFAULT_NAME
```

Comment:
I think you could set `self._name` in the `if user_input is not None:` block from the one they pick since you have the advertisement at that point and drop `DEFAULT_NAME`